base/io.jl

```diff
@@ -174,7 +174,12 @@ Read a single line of text, including a trailing newline character (if one is re
 the end of the input), from the given I/O stream or file (defaults to `STDIN`).
 When reading from a file, the text is assumed to be encoded in UTF-8.
 """
-readline(filename::AbstractString) = open(readline, filename)
+function readline(filename::AbstractString, chomp = false; nl2lf = false)
```

Comment:
Keyword argument currently have a significant performance cost: probably better make `nl2lf` a positional argument.

I'm also not sure we need that argument. Does it exist in other languages? It would seem to me that either you just want to extract the text content of the line and don't care about newline characters (`chomp=true`), or you care about them and you wouldn't want them to be replaced with `\n`.

Finally, arguments must be described in the docstring.